src/Concerns/HasContext.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace BeyondCode\QueryDetector\Concerns;
+
+use Illuminate\Support\Str;
+
+trait HasContext
```

Comment:
The methods are not used anywhere except for the QueryDetector, so a trait does not make a lot of sense to me.

Otherwise, I like the context approach!